src/coprocessor/codec/mysql/json/binary.rs

```diff
@@ -262,8 +262,8 @@ pub trait JsonDecoder: NumberDecoder {
         let mut value_entries_data = &data[key_entries_len..(key_entries_len + value_entries_len)];
         let mut key_offset = key_entries_len + value_entries_len;
         for _ in 0..element_count {
-            let key_real_offset = key_entries_data.decode_u32_le()?;
-            let key_len = key_entries_data.decode_u16_le()?;
+            let key_real_offset = number::decode_u32_le(&mut key_entries_data)?;
```

Comment:
Can we remove all of the old key_entries_data.xxx implementations?

Reply:
Yes, we will remove it step by step @ngaut

Reply:
Could you create a issue to track that?

Reply:
We already have one in internal jira.